lib/matplotlib/widgets.py

```diff
@@ -621,6 +622,247 @@ def disconnect(self, cid):
         except KeyError:
             pass
 
+class TextBox(AxesWidget):
+    """
+    A GUI neutral text input box.
+
+    For the text box to remain responsive
+    you must keep a reference to it.
+
+    The following attributes are accessible
+
+      *ax*
+        The :class:`matplotlib.axes.Axes` the button renders into.
+
+      *label*
+        A :class:`matplotlib.text.Text` instance.
+
+      *color*
+        The color of the text box when not hovering.
+
+      *hovercolor*
+        The color of the text box when hovering.
+
+    Call :meth:`on_text_change` to be updated whenever the text changes
+    Call :meth:`on_submit` to be updated whenever the user hits enter or leaves the text entry field
+    """
+
+    def __init__(self, ax, label, initial = '', 
+                 color='.95', hovercolor='1'):
+        """
+        Parameters
+        ----------
+        ax : matplotlib.axes.Axes
+            The :class:`matplotlib.axes.Axes` instance the button
+            will be placed into.
+
+        label : str
+            Label for this text box. Accepts string.
+        
+        initial : str
+            Initial value in the text box
+            
+        color : color
+            The color of the box
+
+        hovercolor : color
+            The color of the box when the mouse is over it
+        """
+        AxesWidget.__init__(self, ax)
+        
+        self.DIST_FROM_LEFT = .05        
+        
+        self.params_to_disable = []
+        for key in rcParams.keys():                   
+            if u'keymap' in key:
+               self.params_to_disable += [key]
+        
+        self.text = initial
+        self.label = ax.text(0.0,0.5, label,
```

Comment:
PEP8: Run this file through the pep8 tool to catch things like spaces after punctuation, etc.